File: data/src/main/java/com/somename/data/network/LocalApi.java
package com.somename.data.network;


import com.somename.data.BuildConfig;
import com.somename.data.сontent.RootEntity;
import com.somename.data.сontent.RootProductEntity;
import com.somename.data.сontent.RootTopicsEntity;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface LocalApi {

    @GET("posts/all?"+BuildConfig.API_TOKEN+"&search[category]=tech&per_page=10")
    Observable<RootEntity> getTechCategory();

    @GET("topics?search[trending]=true&"+BuildConfig.API_TOKEN+"&per_page=10")
    Observable<RootTopicsEntity> getTrendingTopics();

    @GET("posts/all?"+BuildConfig.API_TOKEN+"&per_page=10")
    Observable<RootEntity> getChosenTopic(@Query("search[topic]") int topicId);

    @GET("posts/{id}?"+BuildConfig.API_TOKEN)
    Observable<RootProductEntity> getProduct(@Path("id") String id);
}

File: data/src/main/java/com/somename/data/network/LocalApi.kt
package com.somename.data.network


import com.somename.data.сontent.RootEntity
import com.somename.data.сontent.RootProductEntity
import com.somename.data.сontent.RootTopicsEntity
import com.somename.domain.BuildConfig

import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface LocalApi {

    @get:GET("posts/all?" + BuildConfig.API_TOKEN + "&search[category]=tech&per_page=10")
    val techCategory: Observable<RootEntity>

    @get:GET("topics?search[trending]=true&" + BuildConfig.API_TOKEN + "&per_page=10")
    val trendingTopics: Observable<RootTopicsEntity>

    @GET("posts/all?" + BuildConfig.API_TOKEN + "&per_page=10")
    fun getChosenTopic(@Query("search[topic]") topicId: Int): Observable<RootEntity>

    @GET("posts/{id}?" + BuildConfig.API_TOKEN)
    fun getProduct(@Path("id") id: String): Observable<RootProductEntity>
}